src/engine/subscription/Subscription.spec.ts

```diff
@@ -0,0 +1,575 @@
+        subscription someSubWithPreProcessorSomeTestsEntityName($input: SomeSubWithPreProcessorSomeTestsEntityNameInput!) {
+          someSubWithPreProcessorSomeTestsEntityName(input: $input) {
+            someTestsEntityName {
+              # someAttribute
+              anotherAttribute
+            }
+          }
+        }`);
+
+      // subscription topic will be automatically generated by shyft
+      let subscription = (await subscribe({
+        schema: graphqlSchema,
+        document: subscriptionDoc,
+        variableValues: {
+          input: {
+            someTestsEntityName: {},
+          },
+        },
+        contextValue: { pubsub },
+      })) as AsyncIterableIterator<any>;
+
+      let pending = subscription.next();
+
+      await pubsub.publish('SomeSubWithPreProcessorSomeTestsEntityName', {
+        // someTestsEntityName: {
+        someAttribute,
+        anotherAttribute: 'world',
+        // },
+      });
+
+      let result = await pending;
+      expect(result).toMatchSnapshot('withoutPreProcessorResult');
+      expect(await subscription.return()).toMatchSnapshot(
+        'withoutPreProcessorEnd',
+      );
+
+      // subscription topic will be generated in preProcessor
+      subscription = (await subscribe({
+        schema: graphqlSchema,
+        document: subscriptionDoc,
+        variableValues: {
+          input: {
+            someTestsEntityName: {
+              someAttribute,
+            },
+          },
+        },
+        contextValue: { pubsub },
+      })) as AsyncIterableIterator<any>;
+
+      pending = subscription.next();
+
+      await pubsub.publish(
+        `SomeSubWithPreProcessorSomeTestsEntityName/${someAttribute}`,
+        {
+          someAttribute,
+          anotherAttribute: 'world',
+        },
+      );
+
+      result = await pending;
+      expect(result).toMatchSnapshot('withPreProcessorResult');
+      expect(await subscription.return()).toMatchSnapshot(
+        'withPreProcessorEnd',
+      );
+    });
+  });
+
+  describe('postProcessor', () => {
+    const testEntity = new Entity({
+      name: 'SomeTestsEntityName',
+      description: 'Just some description',
+      attributes: {
+        someAttribute: {
+          type: DataTypeString,
+          description: 'Just some description',
+          required: false,
+        },
+        anotherAttribute: {
+          type: DataTypeString,
+          description: 'Just some description',
+        },
+      },
+      subscriptions: [
+        new Subscription({
+          name: 'SomeSubWithPostProcessor',
+          type: SUBSCRIPTION_TYPE_CREATE,
+          description: 'build item',
+          attributes: ['someAttribute'],
+          delimiter: '/',
+          postProcessor: (
+            _entity,
+            // _id,
+            _source,
+            input,
+            typeName,
+            entitySubscription,
+            context,
+          ) => {
+            if (context && context.changePayload) {
```

Comment:
With subscription.postProcessor we can override the payload published by Pubsub.